src/Features/Lsif/Generator/Generator.cs

```diff
@@ -335,6 +365,31 @@ void MarkImplementationOfSymbol(ISymbol baseMember)
             return documentVertex.GetId();
         }
 
+        private static async Task GenerateSemanticTokensAsync(
+            Document document,
+            ILsifJsonWriter lsifJsonWriter,
+            IdFactory idFactory,
+            SourceText text,
+            LsifDocument documentVertex)
+        {
+            // Compute colorization data.
+            var data = await SemanticTokensHelpers.ComputeSemanticTokensDataAsync(
+                document,
+                SemanticTokensHelpers.TokenTypeToIndex,
+                new LspProtocol.Range()
+                {
+                    Start = new Position(0, 0),
```

Comment:
i feel like there's a helper which goes from int-pos to lsp-Position.  I'd prefer we use that, and just pass in 0 and text.Length to that helper.

Reply:
@dibarbet do you know of such a thing?

Reply:
Or maybe @allisonchou or @ryanbrandenburg ?

Reply:
@CyrusNajmabadi Are you looking for something like:
```csharp
public static Position AsPosition(this int absoluteIndex, int length){
    return new Position(absoluteIndex, length);
}
```

If so I don't really see what that gets us, if not I don't think I understand the ask.

What might be helpful in this instance (if I'm reading the intention here correctly) is something like
```csharp
public static Range GetRange(this SourceText text)...
```

for retrieving the complete range of a `SourceText`?

Reply:
No, i'm looking for something like `LspPosition GetLspPositionFor(SourceText text, int position)`.  e.g. something to map from the roslyn concept to the lsp concept.  That's where we should centralize this translation, not have code interrogating SourceTexts themselves to try to do the mapping.

Reply:
@CyrusNajmabadi, I found a helper that converts from TextSpan => Range. Is this better?